src/document/DocumentAnnotator.ts

```diff
@@ -60,14 +60,21 @@ export default class DocumentAnnotator extends BaseAnnotator {
         const pageReferenceEl = this.getPageReference(pageEl);
         const managers = this.managers.get(pageNumber) || new Set();
         const resinTags = { fileid: fileId, iscurrent: isCurrentFileVersion };
+        let destroyManagers = false;
 
-        // Destroy any managers that were attached to page elements that no longer exist
+        // If a referenced page element doesn't exist, destroy all managers to keep them in sync
         managers.forEach(manager => {
             if (!manager.exists(pageEl)) {
+                destroyManagers = true;
+            }
```

Comment:
What if the third manager in the list doesn't exist but there are 6 managers. That means the first 2 won't get destroyed. Is that the behavior we want?

Reply:
Ah my mistake, updated. Good catch.